src/ILCompiler.Compiler/src/Compiler/ILStreamReader.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+
+using Internal.TypeSystem;
+using Internal.IL;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.Compiler
+{
+    struct ILStreamReader
```

Comment:
Can you add an XML comment describing what this is? In particular that it doesn't aspire to be a general purpose IL reader.